source/localizable/components/triggering-changes-with-actions.md

```diff
@@ -245,76 +245,90 @@ In this case, the code in `button-with-confirmation` does not change.
 
-```app/templates/components/button-with-confirmation.hbs
-<button {{action "launchConfirmDialog"}}>{{text}}</button>
-{{#if confirmShown}}
-  <div class="confirm-dialog">
-    {{yield confirmValue}}
-    <button class="confirm-submit" {{action "submitConfirm"}}>OK</button>
-    <button class="confirm-cancel" {{action "cancelConfirm"}}>Cancel</button>
-  </div>
-{{/if}}
+```app/components/send-message.js
+export default Ember.Component.extend({
+  actions: {
+    sendMessage(messageType, messageText) {
+      //send message here and return a promise
+    }
+  }
+});
 ```
 
-The `send-message` component provides an input as block content to the `button-with-confirmation` component, setting
-`confirmValue`.
+We want to arrange for the action to be invoked from within `button-with-confirmation` with both arguments.
+A feature of action arguments that applies to this situation is that they _curry_.
+In general (not just in the context of Ember), this refers to the idea that a function with multiple arguments can be expressed as a sequence of functions each taking one argument and returning the next function in the sequence that takes the next argument,
+and so on.
+Evaluating the last function produced in this way is equivalent to evaluating the original function with all of its arguments.
 
-```app/templates/components/send-message.hbs
-{{#button-with-confirmation
-    text="Click to send your message."
-    onConfirm=(action "sendMessage" "info")
-    as |confirmValue|}}
-  {{input value=confirmValue}}
-{{/button-with-confirmation}}
-```
+For our purposes, currying means that when we pass the action to a component, if we provide the action helper with a subset of arguments (e.g. by providing the value of `messageType` when we pass `sendMessage` to `button-with-confirmation`), the result of the helper (in our case `onConfirm`) is a function that can be invoked within the component javascript file with just the remaining arguments,
+in our case `messageText`.
 
-Now when the `submitConfirm` action is invoked, we call it with the value provided by our yielded input.
+Remember, however,
+that our `button-with-confirmation` component does not know or care that it is being used here in a messaging application.
+So to enable it to call the action represented by `onConfirm` with the required remaining argument,
+we will do three things:
 
-```app/components/button-with-confirmation.js
-import Ember from 'ember';
+_First_, in the component we will use a property `confirmValue` to contain a parameter to be passed to `onConfirm`:
 
+```app/components/button-with-confirmation.js
 export default Ember.Component.extend({
   actions: {
-    launchConfirmDialog() {
-      this.set("confirmShown", true);
-    },
-
+    //...
     submitConfirm() {
-      //call onConfirm with the value of the input field as an argument
+      //call onConfirm with a second argument
       const promise = this.get('onConfirm')(this.get('confirmValue'));
```

Comment:
I'd like to shift this to `let` from `const`, but we can address that in a separate PR